mkosi/initrd.py

```diff
@@ -88,89 +89,103 @@ def main() -> None:
-    if args.debug_shell:
-        cmdline += ["--debug-shell"]
-
-    if os.getuid() == 0:
-        cmdline += [
-            "--workspace-dir=/var/tmp",
-            "--package-cache-dir=/var",
-            "--cache-only=metadata",
-        ]
-        if args.format != OutputFormat.directory.value:
-            cmdline += ["--output-mode=600"]
-
-    for d in (
-        "/usr/lib/mkosi-initrd",
-        "/usr/local/lib/mkosi-initrd",
-        "/run/mkosi-initrd",
-        "/etc/mkosi-initrd",
-    ):
-        if Path(d).exists():
-            cmdline += ["--include", d]
-
-    with tempfile.TemporaryDirectory() as d:
-        # Make sure we don't use any of mkosi's default repositories.
-        for p in (
-            "yum.repos.d/mkosi.repo",
-            "apt/sources.list.d/mkosi.sources",
-            "zypp/repos.d/mkosi.repo",
-            "pacman.conf",
+    with tempfile.TemporaryDirectory() as staging_dir:
+        cmdline: list[PathString] = [
+            "mkosi",
+            "--force",
+            "--directory", "",
+            "--format", args.format,
+            "--output", args.output,
+            "--output-dir", staging_dir,
+            "--extra-tree", f"/usr/lib/modules/{args.kernel_version}:/usr/lib/modules/{args.kernel_version}",
+            "--extra-tree=/usr/lib/firmware:/usr/lib/firmware",
+            "--remove-files=/usr/lib/firmware/*-ucode",
+            "--kernel-modules-exclude=.*",
+            "--kernel-modules-include=host",
+            "--build-sources", "",
+            "--include=mkosi-initrd",
+        ]  # fmt: skip
+
+        if args.debug:
+            cmdline += ["--debug"]
+        if args.debug_shell:
+            cmdline += ["--debug-shell"]
+
+        if os.getuid() == 0:
+            cmdline += [
+                "--workspace-dir=/var/tmp",
+                "--package-cache-dir=/var",
+                "--cache-only=metadata",
+            ]
+            if args.format != OutputFormat.directory.value:
+                cmdline += ["--output-mode=600"]
+
+        for d in (
+            "/usr/lib/mkosi-initrd",
+            "/usr/local/lib/mkosi-initrd",
+            "/run/mkosi-initrd",
+            "/etc/mkosi-initrd",
         ):
-            (Path(d) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
-            (Path(d) / "etc" / p).touch()
-
-        # Copy in the host's package manager configuration.
-        for p in (
-            "dnf",
-            "yum.repos.d/",
-            "apt",
-            "zypp",
-            "pacman.conf",
-            "pacman.d/",
-        ):
-            if not (Path("/etc") / p).exists():
-                continue
-
-            (Path(d) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
-            if (Path("/etc") / p).resolve().is_file():
-                shutil.copy2(Path("/etc") / p, Path(d) / "etc" / p)
+            if Path(d).exists():
+                cmdline += ["--include", d]
+
+        with tempfile.TemporaryDirectory() as d:
```

Comment:
Actually this leads to unnecessary indentation, can you merge this with the `with` above to reduce the indentation?